### PR TITLE
Deploy application

### PR DESCRIPTION
**Pull request by divesh987**
### Context

Deploy application

### Changes proposed in this pull request

Add application resource terraform
Update Docker Image Tag in Makefile

### Guidance to review

You can see deployed application advanced health check at https://[getintoteachingapi-development.test.teacherservices.cloud/api/operations/health_check](https://getintoteachingapi-development.test.teacherservices.cloud/api/operations/health_check)

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by divesh987** [comment]
[Deploy application](https://trello.com/c/3Joy2a6A/466-deploy-application)

File: terraform/aks/application.tf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+module "application_configuration" {
+
+  source = "./vendor/modules/aks//aks/application_configuration"
+
+  namespace             = var.namespace
+  environment           = var.environment
+  azure_resource_prefix = var.azure_resource_prefix
+  service_short         = var.service_short
+  config_short          = var.config_short
+  config_variables      =  { PG_CONN_STR = module.postgres.dotnet_connection_string, AKS_ENV_NAME = var.file_environment, REDIS_CONN_STR = module.redis-cache.connection_string, EnableMetrics = false}
```
Comment (thomasleese) [inline terraform/aks/application.tf:35]: These connection strings should be included in the secrets as they contain the credentials needed to connect to the databases.

File: terraform/aks/variables.tf [hunk 2]
```diff
@@ -30,6 +42,18 @@ variable "postgres_flexible_server_sku" { default = "B_Standard_B1ms" }
...
+
 locals {
   azure_credentials = try(jsondecode(var.azure_credentials), null)
+  kv_app_secrets    = yamldecode(data.azurerm_key_vault_secret.app_secrets.value)
+  app_resource_group_name = "${var.azure_resource_prefix}-${var.service_short}-${var.config_short}-rg"
+  app_secrets = merge(
+    local.kv_app_secrets,
+    {
+      DATABASE_URL          = module.postgres.url
+      REDIS_CACHE_URL       = module.redis-cache.url
```
Comment (thomasleese) [inline terraform/aks/variables.tf:56]: Are these necessary if the app is using the connection strings instead?

File: terraform/aks/variables.tf [hunk 3]
```diff
@@ -30,6 +42,18 @@ variable "postgres_flexible_server_sku" { default = "B_Standard_B1ms" }
 variable "postgres_enable_high_availability" { default = false }
 variable "azure_enable_backup_storage" { default = true }
 
+variable "replicas" { default = 1 }
+variable "memory_max" { default = "1Gi" }
+
 locals {
   azure_credentials = try(jsondecode(var.azure_credentials), null)
+  kv_app_secrets    = yamldecode(data.azurerm_key_vault_secret.app_secrets.value)
```
Comment (thomasleese) [inline terraform/aks/variables.tf:50]: The application configuration module already extracts the YAML secrets and sets them as environment variables so I don't think you need this.

File: terraform/aks/config/development_aks.tfvars.json [hunk 4]
```diff
@@ -1,8 +1,10 @@
 {
   "environment": "development",
-  "key_vault_name": "s189t01-gitapi-dv-kv",
-  "key_vault_app_secret_name": "GITAPI-APP-SECRETS-DEVELOPMENT",
-  "key_vault_infra_secret_name": "GIT-INFRA-SECRETS-DEVELOPMENT",
+  "file_environment": "dev",
+  "app_key_vault_name": "s189t01-gitapi-dv-app-kv",
+  "infra_key_vault_name": "s189t01-gitapi-dv-inf-kv",
+  "key_vault_app_secret_name": "API-KEYS",
+  "key_vault_infra_secret_name": "INFRA-KEYS",
```
Comment (thomasleese) [inline terraform/aks/config/development_aks.tfvars.json:7]: If you've got two key vaults (one for the application and one for infrastructure) you shouldn't need a single secret name as the secrets can be added directly to the key vault and extracted using the secrets module:

```tf 
module "infrastructure_secrets" {
  source = "git::https://github.com/DFE-Digital/terraform-modules.git//aks/secrets?ref=stable"

  azure_resource_prefix = var.azure_resource_prefix
  service_short         = var.service_short
  config_short          = var.config_short
  key_vault_short       = "inf"
}

# example: module.secrets.map.STATUSCAKE-TOKEN
```

File: terraform/aks/variables.tf (hunk 2 above)
Comment (divesh987) [reply terraform/aks/variables.tf:56]: Was just following the same standard as register app....will remove

File: terraform/aks/data.tf [hunk 5]
```diff
@@ -0,0 +1,9 @@
+data "azurerm_key_vault" "app_key_vault" {
+  name                = var.app_key_vault_name
+  resource_group_name = local.app_resource_group_name
+}
+
+data "azurerm_key_vault" "infra_key_vault" {
+  name                = var.infra_key_vault_name
+  resource_group_name = local.app_resource_group_name
+}
```
Comment (thomasleese) [inline terraform/aks/data.tf:9]: I think you can remove these (and the associated variables) as they're not being used by anything.

**Review by thomasleese — COMMENTED** [review]
Looks good, just got one comment!

File: terraform/aks/application.tf [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  environment  = var.environment
+  service_name = var.service_name
+
+  cluster_configuration_map = module.cluster_data.configuration_map
+
+  kubernetes_config_map_name = module.application_configuration.kubernetes_config_map_name
+  kubernetes_secret_name     = module.application_configuration.kubernetes_secret_name
+
+
+  docker_image = var.paas_app_docker_image
```
Comment (saliceti) [inline terraform/aks/application.tf:17]: Run terraform fmt to format document

**Review by saliceti — DISMISSED** [review]

File: Makefile [hunk 7]
```diff
@@ -78,46 +79,28 @@ production:
...
-print-infrastructure-secrets: install-fetch-config set-azure-account
-	./fetch_config.rb -s azure-key-vault-secret:${KEY_VAULT}/${INFRASTRUCTURE_SECRETS}  -f yaml
+setup-aks-local-env: set-azure-account set-key-vault-names
+	yaq -i keyvault-secrets:${KEY_VAULT_APPLICATION_NAME} -d yaml -d yaml > GetIntoTeachingApi/env.local
 
-setup-local-env: install-fetch-config set-azure-account
-	./fetch_config.rb -s azure-key-vault-secret:s146d01-local2-kv/${APPLICATION_SECRETS} -f shell-env-var > GetIntoTeachingApi/env.local
 
-terraform-init: install-terrafile set-azure-account
+terraform-init: set-azure-account
```
Comment (saliceti) [inline Makefile:95]: This needs `bin/terrafile` first

**Review by saliceti — APPROVED** [review]

**Pull request merged**